src/typedefs.jl

```diff
@@ -42,6 +42,11 @@ struct LevelsException{T, R} <: Exception
     levels::Vector{T}
 end
 
+struct OrderedLevelsException{T} <: Exception
+    newlevel::Any
```

Comment:
Since you added a type parameter for `levels`, maybe add one for `newlevel` too? Not a big deal since that's for errors, but...

Reply:
[`convert(T, level)`](https://github.com/JuliaData/CategoricalArrays.jl/pull/125/commits/b53740cb4a048148711b2ef8bc902781baa0df13#diff-a443ccbb96d26cdb0187b6276e1e4170R96) is called after the exception is [thrown](https://github.com/JuliaData/CategoricalArrays.jl/pull/125/commits/b53740cb4a048148711b2ef8bc902781baa0df13#diff-a443ccbb96d26cdb0187b6276e1e4170R113) so `newlevel` might not be of type `T`.

Reply:
Yeah, I meant adding a separate parameter.